tests/jenkins/cloud-native-dev-with-spring-boot.groovy

```diff
@@ -0,0 +1,308 @@
+                          --server '${imap_server}' \
+                          --guid ${guid} \
+                          --timeout 40 \
+                          --filter 'has completed'
+                        """
+                    ).trim()
+
+                    try {
+                        def m = email =~ /Connect to the shared cluster by pointing your web browser to <a href="(https:\/\/master\.[^"]+)/
+                        openshift_location = m[0]
+                        echo "openshift_location = '${openshift_location}'"
+
+                    } catch(Exception ex) {
+                        echo "Could not parse email:"
+                        echo email
+                        echo ex.toString()
+                        throw ex
+                    }
+                }
+            }
+        }
+
+        stage('Confirm before retiring') {
+            when {
+                expression {
+                    return params.confirm_before_delete
+                }
+            }
+            steps {
+                input "Continue ?"
+            }
+        }
+        stage('Retire service from CF') {
+            environment {
+                uri = "${cf_uri}"
+                credentials = credentials("${opentlc_creds}")
+                admin_credentials = credentials("${opentlc_admin_creds}")
+                DEBUG = 'true'
+            }
+            /* This step uses the delete_svc_guid.sh script to retire
+             the service from CloudForms */
+            steps {
+                git 'https://github.com/redhat-gpte-devopsautomation/cloudforms-oob'
+
+                sh "./opentlc/delete_svc_guid.sh '${guid}'"
+            }
+            post {
+                failure {
+                    withCredentials([usernameColonPassword(credentialsId: imap_creds, variable: 'credentials')]) {
+                        mail(
+                            subject: "${env.JOB_NAME} (${env.BUILD_NUMBER}) failed retiring for GUID=${guid}",
+                            body: "It appears that ${env.BUILD_URL} is failing, somebody should do something about that.\nMake sure GUID ${guid} is destroyed.",
+                            to: "${notification_email}",
+                            replyTo: "${notification_email}",
+                            from: credentials.split(':')[0]
+                        )
+                    }
+                    withCredentials([string(credentialsId: rocketchat_hook, variable: 'HOOK_URL')]) {
+                        sh(
+                            """
+                            curl -H 'Content-Type: application/json' \
+                            -X POST '${HOOK_URL}' \
+                            -d '{\"username\": \"jenkins\", \"icon_url\": \"https://dev-sfo01.opentlc.com/static/81c91982/images/headshot.png\", \"text\": \"@here :rage: ${env.JOB_NAME} (${env.BUILD_NUMBER}) failed retiring ${guid}.\"}'\
+                            """.trim()
+                        )
```

Comment:
I think we can get rid of the rocketchat all together since nobody use it anymore

Even better: write the equivalent for google chat.

Reply:
I will try to do that after this agnosticv conversion.